internal/jira/jira_boards.go

```diff
@@ -12,19 +12,33 @@ const (
 
 type findBoardsQueryParams struct {
 	ProjectKeyOrId string `url:"projectKeyOrId"`
+	StartAt        int    `url:"startAt"`
 }
 
 func (api *httpApi) FindBoards(projectKeyOrId string) ([]BoardItem, error) {
-	resultBytes, err := api.jiraRequest("GET", FindAllBoardsUrl, &findBoardsQueryParams{ProjectKeyOrId: projectKeyOrId}, nil)
-	if err != nil {
-		return nil, err
-	}
-	var result BoardsResponse
-	err = json.Unmarshal(resultBytes, &result)
-	if err != nil {
-		return nil, err
+	params := &findBoardsQueryParams{ProjectKeyOrId: projectKeyOrId, StartAt: 0}
+	var boards []BoardItem
+	for {
```

Comment:
in theory ... that loop could run infinite number of time. It should not happen of course - but in theory is possible. What do you think about some limitation? like "max-rounds" "max-depth" "max-iterations"? just to break that for loop .. after N iterations.

Reply:
seems pointlessly defensive
what kind of error are you going to throw when the jira API breaks its pagination contract?

Reply:
any kind of error is better then infinitive loop, right? 😄

Reply:
but in general you are right - it seems to be pointlessly defensive. All right, I'm gonna to merge it